variables.tf

```diff
@@ -303,3 +303,24 @@ variable "cluster_autoscaler_helm_config" {
   description = "Cluster Autoscaler Helm Config"
   default     = { "version" : "9.28.0" }
 }
+
+variable "enable_ecr_pullthrough_rule" {
+  type        = bool
+  description = "Specifies if ECR pull-through rule, and accompanying resources, will be created."
+  default     = false
+}
+
+variable "enable_ecr_pullthrough_secret" {
+  type        = bool
+  description = "Specifies if secret storing ECR pull-through credentials will be created."
+  default     = false
+}
+
+variable "registry_credentials" {
+  type        = map(string)
```

Comment:
Why don't you be more specific and inside the map use a object?
I take it we'll always expect `username` and `accessToken` for `registry_credentials`.
Therefore, it'd be in our best interest to use a defined object like so.

Reply:
For this particular registry, this could be considered an overkill, since it does not require any authentication at all.  
Generally, yes, it's always `username `and `accessToken`, but there's discussion below with SimonG to have this hardcoded instead, for this particular registry...

Reply:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret_version#key-value-pairs

Reply:
If `registry_credentials` variable is agreed upon, I would urge that you map an object to the proposed attributes in order to prevent the injection of any unexpected/harmful key-value pairs.

Reply:
Agreements can be re-agreed, one agreement should not affect 999 others :smiley:  
In any case, this was copied from upstream, I understand there is possibility for end user to inject unwanted kv pair(s), but what you propose is just a measure to prevent end-user shooting themselves in the foot.  
Waiting for response from SimonG (below; it's connected to your input as well, because he wants it hard-coded).